libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PathLookupImpl.java

```diff
@@ -0,0 +1,147 @@
+        requireNonNull(tempDir);
+        requireNonNull(settingResolver);
+    }
+
+    @Override
+    public Stream<Path> getBaseDirPaths(BaseDir baseDir) {
+        if (baseDir == BaseDir.HOME) {
+            return Stream.of(homeDir);
+        } else if (baseDir == BaseDir.DATA) {
+            return Arrays.stream(dataDirs);
+        } else if (baseDir == BaseDir.SHARED_REPO) {
+            return Arrays.stream(sharedRepoDirs);
+        } else if (baseDir == BaseDir.CONFIG) {
+            return Stream.of(configDir);
+        } else if (baseDir == BaseDir.LIB) {
+            return Stream.of(libDir);
+        } else if (baseDir == BaseDir.MODULES) {
+            return Stream.of(modulesDir);
+        } else if (baseDir == BaseDir.PLUGINS) {
+            return Stream.of(pluginsDir);
+        } else if (baseDir == BaseDir.LOGS) {
+            return Stream.of(logsDir);
+        } else if (baseDir == BaseDir.TEMP) {
+            return Stream.of(tempDir);
+        } else if (baseDir == BaseDir.PID) {
+            return pidFile == null ? null : Stream.of(pidFile);
+        } else {
+            throw new IllegalStateException("unknown base dir [" + baseDir + "]");
+        }
+    }
+
+    @Override
+    public Stream<Path> resolveRelativePaths(BaseDir baseDir, Path relativePath) {
+        if (baseDir == BaseDir.HOME) {
+            return Stream.of(homeDir.resolve(relativePath));
+        } else if (baseDir == BaseDir.DATA) {
+            return Arrays.stream(dataDirs).map(dataDirPath -> dataDirPath.resolve(relativePath));
+        } else if (baseDir == BaseDir.SHARED_REPO) {
+            return Arrays.stream(sharedRepoDirs).map(sharedRepoDir -> sharedRepoDir.resolve(relativePath));
+        } else if (baseDir == BaseDir.CONFIG) {
+            return Stream.of(configDir.resolve(relativePath));
+        } else if (baseDir == BaseDir.LIB) {
+            return Stream.of(libDir.resolve(relativePath));
+        } else if (baseDir == BaseDir.MODULES) {
+            return Stream.of(modulesDir.resolve(relativePath));
+        } else if (baseDir == BaseDir.PLUGINS) {
+            return Stream.of(pluginsDir.resolve(relativePath));
+        } else if (baseDir == BaseDir.LOGS) {
+            return Stream.of(logsDir.resolve(relativePath));
+        } else if (baseDir == BaseDir.TEMP) {
+            return Stream.of(tempDir.resolve(relativePath));
+        } else if (baseDir == BaseDir.PID) {
+            return pidFile == null ? null : Stream.of(pidFile.resolve(relativePath));
+        } else {
+            throw new IllegalStateException("unknown base dir [" + baseDir + "]");
+        }
+    }
+
+    @Override
+    public Stream<Path> resolveSettingPaths(BaseDir baseDir, String settingName) {
+        Stream<Path> relativePaths = settingResolver.apply(settingName)
+            .filter(s -> s.toLowerCase(Locale.ROOT).startsWith("https://") == false)
+            .distinct()
+            .map(Path::of);
+        if (baseDir == BaseDir.HOME) {
```

Comment:
Switch would be nicer here. It's unfortunate the combination is inverted. While we could reuse getBaseDirPaths, we lose the knowledge there is only a single base dir, and so would have to do the paths combination as with data dirs...